runtime/src/snapshot_utils.rs

```diff
@@ -395,6 +396,25 @@ pub fn archive_snapshot_package(
         ("duration_ms", timer.as_ms(), i64),
         ("size", metadata.len(), i64)
     );
+
+    info!("Create snapshot local file in {:?}", tar_dir);
+    create_snapshot_local_file(tar_dir)?;
+    Ok(())
+}
+
+pub fn has_snapshot_local_file(snapshot_archive_dir: &Path) -> bool {
+    snapshot_archive_dir.join(LOCAL_SNAPSHOT_FILE_NAME).exists()
+}
+
+pub fn create_snapshot_local_file(snapshot_archive_dir: &Path) -> Result<()> {
+    let _local_file = File::create(snapshot_archive_dir.join(LOCAL_SNAPSHOT_FILE_NAME))?;
+    Ok(())
+}
+
+pub fn delete_snapshot_local_file(snapshot_archive_dir: &Path) -> Result<()> {
+    if has_snapshot_local_file(snapshot_archive_dir) {
+        fs::remove_file(snapshot_archive_dir.join(LOCAL_SNAPSHOT_FILE_NAME))?;
+    }
```

Comment:
I dunno how much it matters/how likely this is, but there's a race condition here; the file could have been deleted after `has_local_snapshot_file()` returns true. Maybe this fn should call only `remove_file()` and then swallow the "file does not exist" error.